$OpenBSD: patch-backend_usb-libusb_c,v 1.13 2014/12/10 10:22:13 ajacoutot Exp $

commit ce7d367df7e8b554116e297d6e38a9b989dea725
Author: msweet <msweet@a1ca3aef-8c08-0410-bb20-df032aa958be>
Date:   Tue Dec 9 21:33:23 2014 +0000
Fix a libusb issue on OpenBSD.

commit 5e1504aa1e8854f222154d627148ad5675dc7de3
Author: msweet <msweet@a1ca3aef-8c08-0410-bb20-df032aa958be>
Date:   Tue Dec 9 22:10:45 2014 +0000
Fix USB printing on OpenBSD (STR #4525)

--- backend/usb-libusb.c.orig	Wed Dec 10 10:48:01 2014
+++ backend/usb-libusb.c	Wed Dec 10 10:48:18 2014
@@ -915,7 +915,7 @@ find_device(usb_cb_t   cb,		/* I - Callback function *
 		  write_endp = endp;
 	      }
 
-            if (write_endp >= 0)
+            if (write_endp != 0xff)
 	    {
 	     /*
 	      * Save the best match so far...
@@ -1451,9 +1451,14 @@ open_device(usb_printer_t *printer,	/* I - Printer */
   else
   {
     printer->usblp_attached = 0;
-    fprintf(stderr, "DEBUG: Failed to check whether %04x:%04x has the \"usblp\" kernel module attached\n",
-	      devdesc.idVendor, devdesc.idProduct);
-    goto error;
+
+    if (errcode != LIBUSB_ERROR_NOT_SUPPORTED)
+    {
+      fprintf(stderr,
+              "DEBUG: Failed to check whether %04x:%04x has the \"usblp\" "
+              "kernel module attached\n", devdesc.idVendor, devdesc.idProduct);
+      goto error;
+    }
   }
 
  /*
